ons_alpha/taxonomy/models.py

```diff
@@ -0,0 +1,98 @@
+from django import forms
+from django.core.exceptions import ValidationError
+from django.db import models
+from treebeard.mp_tree import MP_Node
+from wagtail.admin.forms import WagtailAdminModelForm
+from wagtail.admin.panels import FieldPanel
+from wagtail.search import index
+
+
+# This is the main 'node' model, it inherits mp_node
+# mp_node is short for materialized path, it means the tree has a clear path
+class Topic(index.Indexed, MP_Node):
+    name = models.CharField(max_length=100)
+
+    node_order_by = ["name"]
+
+    panels = [
+        FieldPanel("name", classname="full"),
+        FieldPanel(
+            "parent", help_text="Where this topic will be nested under. Leave blank for top-level topics."
+        ),  # parent is not a field on the model, it is built in the TopicForm form class
+    ]
+
+    search_fields = [
+        index.FilterField("name"),
+        index.FilterField("depth"),
+        index.SearchField("name"),
+        index.AutocompleteField("name"),
+    ]
+
+    def __str__(self):
+        return self.name_with_depth()
+
+    # this is just a convenience function to make the names appear with lines
+    # eg root | - first child
+    def name_with_depth(self):
```

Comment:
Side note: Wagtail's Collection object has a method that does the indentation - https://github.com/wagtail/wagtail/blob/2d568dd825e4b087d7cea1c4d1fa49d916b395c2/wagtail/models/media.py#L93 - one we could use in the future.

This works fine for our needs